lib/applications.js

```diff
@@ -54,10 +65,41 @@ class Application {
 //   applications - array of JSTP applications
 //
 const createAppsIndex = (applications) => {
-  const index = {};
+  const latestApps = new Map();
+  const index = new Map();
 
   applications.forEach((application) => {
-    index[application.name] = application;
+    let appVersions = index.get(application.name);
+    if (!appVersions) {
+      appVersions = new Map();
+      index.set(application.name, appVersions);
+    }
+    if (!application.version) {
+      // no version means latest version
+      if (appVersions.has('latest')) {
+        throw new Error(
+          `Multiple entries of '${application.name} without version`
+        );
+      } else {
+        appVersions.set('latest', application);
+      }
+    } else {
+      appVersions.set(application.version, application);
+
+      // save latest version to fill missing latest versions later
+      const latestApp = latestApps.get(application.name);
+      if (!latestApp || semver.gt(application.version, latestApp.version)) {
+        latestApps.set(application.name, application);
+      }
+    }
+  });
+
+  // set latest versions of apps without explicit latest version
+  latestApps.forEach((application, appName) => {
+    const appVersions = index.get(appName);
+    if (!appVersions.get('latest')) {
```

Comment:
You should use `has` method here instead of `get` because you don't need the value, just checking if it exists.